ext/fetch/lib.rs

```diff
@@ -405,18 +408,12 @@ pub trait FetchPermissions {
     api_name: &str,
   ) -> Result<(), PermissionCheckError>;
   #[must_use = "the resolved return value to mitigate time-of-check to time-of-use issues"]
-  fn check_read<'a>(
-    &mut self,
-    path: Cow<'a, Path>,
-    api_name: &str,
-    get_path: &'a dyn deno_fs::GetPath,
-  ) -> Result<deno_fs::CheckedPath<'a>, FsError>;
-  fn check_write<'a>(
+  fn check_open<'a>(
```

Comment:
Replaced all the `check_read`/`check_write`-like methods with only `check_open` and then you specify on that if you want Read/Write/Both permissions. This should help a bit with performance and make things simpler.